src/var.jl

```diff
@@ -42,18 +42,18 @@ macro ncpolyvar(args...)
 end
 
 struct PolyVar{C} <: AbstractVariable
-    id::Int
+    id::UInt
```

Comment:
What's the advantage of using `UInt` here ?

Reply:
I was just wanting to save the hash of Sym here. I can switch it back to Int and bitcast the hash too... I'm realizing that I'll need something better than hash to avoid collision. Also would you be open to check both fields in isequal? Or is that performance critical overhead?

Reply:
Yes, equality testing between variables is in most hot loops for performance. Maybe SymbolicUtils can check for collisions when it generates the `id`s ? Similarly to how `Dict` handles collisions between hashes. It's best to handle collision just once at construction in SymbolicUtils that everytime we will compare two variables

Reply:
Yeah I guessed that would be the case. The thing is to do hash collision detection we would need a global dictionary... and weakkeydicts will not work because things are immutable. We just need a way to assign a unique number to a symbol. I think maybe we could use a bloom filter to check if a new has definitely has not been used, if it is possibly used then we could just increment the hash and repeat. The storage for that should be way smaller.

Reply:
I used BloomFilters, so we're good! https://github.com/JuliaSymbolics/SymbolicUtils.jl/pull/231/commits/a0f5a2fb9e272d1d735af1493afdc946191c298c

Merge this PR if you think it's okay now! Thank you!